tests/bson/bson-decimal128interface-002.phpt

```diff
@@ -0,0 +1,97 @@
+--TEST--
+MongoDB\BSON\Decimal128Interface example with MongoDB\BSON\TypeWrapper
+--FILE--
+<?php
+
+require_once __DIR__ . '/../utils/tools.php';
+
+class MyDecimal128 implements MongoDB\BSON\TypeWrapper, MongoDB\BSON\Decimal128Interface
+{
+    private $decimal;
+
+    public function __construct($value)
+    {
+        $this->decimal = new MongoDB\BSON\Decimal128($value);
+    }
+
+    public static function createFromBSONType(MongoDB\BSON\Type $type)
+    {
+        if ( ! $type instanceof MongoDB\BSON\Decimal128) {
+            throw new InvalidArgumentException('Cannot create MyDecimal128 from ' . get_class($type));
+        }
+
+        return new self($type);
+    }
+
+    public function toBSONType()
+    {
+        return $this->decimal;
+    }
+
+    public function __toString()
+    {
+        return (string) $this->decimal;
+    }
+}
+
+$decimal = new MongoDB\BSON\Decimal128('1234.5678');
+$myDecimal = new MyDecimal128('1234.5678');
+
+echo "\nTesting MyDecimal128 and Decimal128 are equivalent\n";
+var_dump((string) $decimal === (string) $myDecimal);
```

Comment:
We discussed this while doing the HHVM versions, and this is what the conclusion was I believe: https://github.com/mongodb/mongo-hhvm-driver/blob/master/ext_mongodb.php#L1182